File: Examples/test-suite/javascript/smart_pointer_not_runme.js
var smart_pointer_not = require("smart_pointer_not");

f = new smart_pointer_not.Foo();
b = new smart_pointer_not.Bar(f);
s = new smart_pointer_not.Spam(f);
g = new smart_pointer_not.Grok(f);

// This is the only that should work
f.getx();

// Accessing an inexistent property in JS
// does not throw - it returns undefined
fail = false;
try {
    x = b.getx();
    fail = new Error("Error! b.x");
} catch { }
if (fail) throw fail;

try {
    x = s.getx();
    fail = new Error("Error! b.x");
} catch { }
if (fail) throw fail;

try {
    x = g.getx();
    fail = new Error("Error! b.x");
} catch { }
if (fail) throw fail;

File: Examples/test-suite/lua/smart_pointer_not_runme.lua
require("import")
require("smart_pointer_not")
spn=smart_pointer_not
catch_undef_globs() -- catch "undefined" global variables

foo = spn.Foo()
foo.x = 7
assert(foo:getx() == 7)

bar = spn.Bar(foo)
success = pcall(bar.getx, bar) -- Bar is not a smart pointer. Call should fail
assert(not success)

spam = spn.Spam(foo)
success = pcall(spam.getx, spam) -- Spam is not a smart pointer. Call should fail
assert(not success)

grok = spn.Grok(foo)
success = pcall(grok.getx, grok) -- Spam is not a smart pointer. Call should fail
assert(not success)
